tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go

```diff
@@ -0,0 +1,229 @@
+			err := gitdetails.SetGitDetailsInArgoCd(
+				tsparams.ArgoCdClustersAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName],
+				true, false)
+			Expect(err).ToNot(HaveOccurred(), "Failed to reset clusters app git details")
+
+			// Test teardown expected results validation.
+			By("checking the infra env manifests exists on hub")
+			_, err = assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke infra env manifests")
+
+			By("checking the bare metal host manifests exists on hub")
+			_, err = bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke bmh manifests")
+
+			By("checking the cluster deployment manifests exists on hub")
+			_, err = hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke cluster deployment manifests")
+
+			By("checking the NM state config manifests exists on hub")
+			nmStateConfigList, err := assisted.ListNmStateConfigs(HubAPIClient, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to list NM state config manifests")
+			Expect(nmStateConfigList).ToNot(BeEmpty(), "Failed to find NM state config manifests")
+
+			By("checking the klusterlet addon config manifests exists on hub")
+			_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke kac manifests")
+
+			By("checking the agent cluster install manifests exists on hub")
+			_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke agent cluster install manifests")
+		})
+
+		// 75374 - Detaching the AI multi-node openshift (MNO) spoke cluster.
+		It("Validate detaching the AI multi-node openshift spoke cluster", reportxml.ID("75374"), func() {
+			By("checking spoke cluster type")
+			spokeClusterType, err := rancluster.CheckSpokeClusterType(RANConfig.Spoke1APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to fetch spoke cluster type")
+
+			if spokeClusterType == ranparam.SNOCluster {
+				Skip("This test only applies to standard or multi-node openshift spoke cluster")
+			}
+
+			// Test step 1-Delete default assisted installer template reference ConfigMap CRs after spoke cluster installed.
+			By("deleting default assisted installer template reference ConfigMap custom resources", func() {
+
+				By("deleting default assisted installer cluster level templates ConfigMap CR")
+				clusterTemplateConfigMap, err := configmap.Pull(HubAPIClient, tsparams.DefaultAIClusterTemplatesConfigMapName,
+					ranparam.AcmOperatorNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to find default AI cluster level templates config map")
+
+				if err == nil {
```

Comment:
since you assert the error didn't occur, it is always `nil` so the if statement is unnecessary. Is it a test failure if the cm does not exist?

Reply:
Is it a test failure if the cm does not exist? **re: `no, i am deleting default cluster level templates ConfigMap custom resource after spoke cluster installed, and then make sure the spoke cluster is still accessible`**

Reply:
since you assert the error didn't occur, it is always nil so the if statement is unnecessary. re: `I am deleting ConfigMap custom resource after asserting err == nil i.e. first I am trying to find default cluster level templates ConfigMap customer resource exists in rhacm namespace, and then delete it if it exists`

Reply:
@mpmaruthu In that case then, I think you want to remove the `Expect` line since that will fail the test if the resource does not exist. This goes for all of the places where you're trying to delete if it does not exist

Reply:
updated. Thanks!